docs/reference/docs/delete.asciidoc

```diff
@@ -4,18 +4,19 @@
 <titleabbrev>Delete</titleabbrev>
 ++++
 
-Removes a JSON document from the specified index.
+Removes a JSON document from a specified data stream or index.
```

Comment:
We disallow deletes directly on a data stream because it would require searching every backing index for the specified ID. If a document is to be deleted from a data stream, the relevant backing index must be targeted directly.

Reply:
:doh: Thanks for catching this. I think I got overzealous. 

Fixed up here: https://github.com/elastic/elasticsearch/pull/58340/files#diff-907ef82eb2dfb22151e4312359f5aee4